swift_browser_ui/ui/middlewares.py

```diff
@@ -27,6 +29,21 @@ def return_error_response(error_code: int) -> web.Response:
         )
 
 
+@web.middleware
+async def check_session_at(
+    request: web.Request,
+    handler: AiohttpHandler,
+) -> web.Response:
+    """Raise on expired sessions."""
+    session = await aiohttp_session.get_session(request)
+    if "at" in session:
```

Comment:
we could also check the token and projects are present, or having a valid session is enough ?

Reply:
Valid session is enough, since the login process can't finish without the user having at least one project available, and a successful login to that project. That actually reminds me I need to remove some additional checks from the frontend.